### PR TITLE
Topic/single activtiy app

### PR DESCRIPTION
**Pull request by Jaber-aldroubi**

### CODE REVIEW

File: app/src/main/java/com/example/amcc/view/MainFragment.java [hunk 1]
```diff
@@ -72,9 +80,18 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
 
         navController = Navigation.findNavController(view);
         view.findViewById(R.id.btnResultID).setOnClickListener(v -> {
-            CarDetails car = new CarDetails(citySelected, Integer.parseInt(engineSizeField.getText().toString()), Integer.parseInt(emissionEdtField.getText().toString()), FuelType.e5, regDate, Integer.parseInt(avgConField.getText().toString()), Integer.parseInt(milePerYField.getText().toString()));
-            viewModel.setApiData(car);
-            navController.navigate(R.id.resultFragment);
+
+            validateInput();
+            if (validation.validate()) {
```
Comment (Shaftar) [inline app/src/main/java/com/example/amcc/view/MainFragment.java]: @Jaber-aldroubi  here is duplicated code. Why do you have tow times validation? validateInput() could been return method. you should use one of them?

File: app/src/main/java/com/example/amcc/view/MainFragment.java [hunk 2]
```diff
@@ -72,9 +80,18 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+                CarDetails car = new CarDetails(citySelected,
+                        Integer.parseInt(engineSizeField.getText().toString()),
+                        Integer.parseInt(emissionEdtField.getText().toString()),
+                        FuelType.e5, regDate,
```
Comment (Shaftar) [inline app/src/main/java/com/example/amcc/view/MainFragment.java]: @Jaber-aldroubi I should say, FuelType is useless! String variable should be enough, bec. we have fixed RadioButton  input.

File: app/src/main/java/com/example/amcc/view/MainFragment.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        spinner = view.findViewById(R.id.spinnerCityHolderID);
+        emissionEdtField = view.findViewById(R.id.edtNumEmissionID);
+        engineSizeField = view.findViewById(R.id.edtNumEngSizeID);
+        avgConField = view.findViewById(R.id.edtNumConsumeID);
+        milePerYField = view.findViewById(R.id.edtNumMileAgeYearID);
+        calendarView = view.findViewById(R.id.calendarViewID);
```
Comment (Shaftar) [inline app/src/main/java/com/example/amcc/view/MainFragment.java]: @Jaber-aldroubi  What does calenderView Here?

File: app/src/main/java/com/example/amcc/view/MainFragment.java [hunk 4]
```diff
@@ -54,47 +54,48 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+            emissionEdtField = view.findViewById(R.id.edtNumEmissionID);
+            engineSizeField = view.findViewById(R.id.edtNumEngSizeID);
+            avgConField = view.findViewById(R.id.edtNumConsumeID);
+            milePerYField = view.findViewById(R.id.edtNumMileAgeYearID);
+            regDateField = view.findViewById(R.id.edtFirst_reg_year);
+
+            radioGroup = view.findViewById(R.id.fuel_type_radio_group);
+            int radioId = radioGroup.getCheckedRadioButtonId();
+            radioButton = view.findViewById(radioId);
+            String fuelType = radioButton.getText().toString();
```
Comment (Shaftar) [inline app/src/main/java/com/example/amcc/view/MainFragment.java]: @Jaber-aldroubi Here you have big mistake! What value does fuelType return?

Comment (Jaber-aldroubi) [reply app/src/main/java/com/example/amcc/view/MainFragment.java]: fuelType return the string, which the user choose (E10, Super, Diesel)

Comment (Jaber-aldroubi) [reply app/src/main/java/com/example/amcc/view/MainFragment.java]: Thy will be later converted to e10, e5, diesel. Case letter is also important

File: app/src/main/java/com/example/amcc/view/MainFragment.java (hunk 3 above)
Comment (Jaber-aldroubi) [reply app/src/main/java/com/example/amcc/view/MainFragment.java]: @Shaftar yes, should be deleted

File: app/src/main/java/com/example/amcc/view/MainFragment.java (hunk 1 above)
Comment (Jaber-aldroubi) [reply app/src/main/java/com/example/amcc/view/MainFragment.java]: validation is an object from Validation Class (Library ). validate() is a function in this class which will return (true or false) depends on validateInput, whichs return nothing at all.

Comment (Shaftar) [reply app/src/main/java/com/example/amcc/view/MainFragment.java]: I mean this method should return validation.validate(); So you have to check if (validateInput())

Comment (Jaber-aldroubi) [reply app/src/main/java/com/example/amcc/view/MainFragment.java]: Reworked

**Review by Shaftar — APPROVED** [review]

File: app/src/main/java/com/example/amcc/retrofitApi/RetrofitClient.java [hunk 5]
```diff
@@ -2,18 +2,28 @@
...
-    private final String BASE_Url = "https://amcosts.azurewebsites.net/";
+    private final String BASE_Url = "http://77.23.21.154/amcc_api/";
+
+
     private AmccApi amccApi;
-    private Retrofit retrofit;
 
     private RetrofitClient() {
-        retrofit = new Retrofit.Builder()
+        Gson gson = new GsonBuilder()
```
Comment (Shaftar) [inline app/src/main/java/com/example/amcc/retrofitApi/RetrofitClient.java:22]: @Jaber-aldroubi what does that?

File: app/src/main/java/com/example/amcc/viewModel/SharedViewModel.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                Log.d(TAG, "onFailure: " + t.getMessage());
+            }
+        });
+    }
+
+    public void setCitiesArray() {
+        client.getCities().enqueue(new Callback<List<String>>() {
+            @Override
+            public void onResponse(Call<List<String>> call, Response<List<String>> response) {
+                System.out.println("fuck you bitch:" + response.code());
```
Comment (Shaftar) [inline app/src/main/java/com/example/amcc/viewModel/SharedViewModel.java:51]: @Jaber-aldroubi please don't forget to replace with LiveDate Module.

**Review by Shaftar — APPROVED** [review]

**Review by Shaftar — APPROVED** [review]

**Pull request merged**